daemon/src/taker_cfd.rs

```diff
@@ -167,6 +189,71 @@ impl<O, M, W> Actor<O, M, W> {
 
         Ok(())
     }
+
+    async fn handle_auto_rollover(&mut self) -> Result<()> {
+        tracing::debug!("Auto rollover");
+
+        let mut conn = self.db.acquire().await?;
+        let cfds = load_all_cfds(&mut conn).await?;
+
+        // cleanup all pending proposals because auto-rollover will create a new one
+        self.current_pending_proposals.clear();
+
+        // Only cfds in state `Open` are allowed for auto-rollover proposal
+        for cfd in cfds
+            .iter()
+            .filter(|cfd| matches!(cfd.state, CfdState::Open { .. }))
+        {
+            tracing::debug!(
+                "settlement_interval: {:?}",
+                self.roll_over_params.settlement_interval
+            );
+
+            let now = OffsetDateTime::now_utc();
+            let elapsed_since_creation = now - cfd.creation_timestamp()?;
+
+            // trigger rollover if rollover window has elapsed
+            if cfd.expiry_timestamp() > now
+                && elapsed_since_creation
+                    > self.roll_over_params.rollover_offset
+                        * self.roll_over_params.settlement_interval
+            {
+                tracing::debug!("proposed roll over");
+                try_continue!(self.propose_roll_over(cfd.order.id).await)
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn propose_roll_over(&mut self, order_id: OrderId) -> Result<()> {
+        if self.current_pending_proposals.contains_key(&order_id) {
+            anyhow::bail!("An update for order id {} is already in progress", order_id)
+        }
```

Comment:
not currently, but if someones makes `propose_rollover` pub in the future, it would be nice to have this check. I dont think it does any harm to keep it?

Reply:
It does harm. It introduces complexity because its presence makes you think that it can be hit. Unnecessary code is one of the worst forms of complexity because you end up wasting hours of looking at it to only realize that is unrelated to the problem you are debugging.